### PR TITLE
feat(cellguide): implement pipeline for the source collections data artifact

### PR DESCRIPTION
**Pull request by joyceyan**
## Reason for Change

https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/5466

## Changes

- generates the source collections data artifact
- i added a new citation formatter to format responses from the collections API

## Testing steps

- added a test for generating source collections data
- i also added a test for formatting citations from the collections API

### CODE REVIEW

File: backend/cellguide/pipeline/canonical_marker_genes/utils.py
Comment (atarashansky) [inline backend/cellguide/pipeline/canonical_marker_genes/utils.py:1]: Woah, didn't know there were so many variations for the different pub metadata!

Comment (atarashansky) [reply backend/cellguide/pipeline/canonical_marker_genes/utils.py:1]: Oh, I just realized you're overloading the same function to format citations from both crossref and the data portal.

Comment (atarashansky) [reply backend/cellguide/pipeline/canonical_marker_genes/utils.py:1]: I think it would be better to split these out into `format_citation_crossref` and `format_citation_dp` to be explicit about what kind of publisher metadata each function expects.

File: backend/cellguide/pipeline/source_collections/source_collections_generator.py
Comment (atarashansky) [inline backend/cellguide/pipeline/source_collections/source_collections_generator.py:46]: See above comment about splitting `format_citation` into two functions for crossref and data portal publisher metadata, respectively:

> I think it would be better to split these out into format_citation_crossref and format_citation_dp to be explicit about what kind of publisher metadata each function expects.

File: backend/cellguide/pipeline/source_collections/source_collections_generator.py
Comment (atarashansky) [inline backend/cellguide/pipeline/source_collections/source_collections_generator.py:66]: out of curiosity, have you observed duplicate diseases/tissues/organisms in the source data or is this just a guardrail?

File: tests/unit/cellguide_pipeline/test_canonical_marker_genes.py
Comment (atarashansky) [inline tests/unit/cellguide_pipeline/test_canonical_marker_genes.py:83]: nice :)

**Review by atarashansky — APPROVED** [review]
Approving now - I'll leave it up to you if you want to split format_citation into two functions or not. It's just a nit from me.

File: tests/unit/cellguide_pipeline/test_source_collections_generator.py
Comment (atarashansky) [inline tests/unit/cellguide_pipeline/test_source_collections_generator.py:16]: typo: test__source
```suggestion
    def test__source_collections_generator(self):
```

File: backend/cellguide/pipeline/canonical_marker_genes/utils.py
Comment (joyceyan) [reply backend/cellguide/pipeline/canonical_marker_genes/utils.py:1]: ohhh okay this makes so much more sense. i was wondering why the response data was so different 🤔

File: backend/cellguide/pipeline/source_collections/source_collections_generator.py
Comment (joyceyan) [reply backend/cellguide/pipeline/source_collections/source_collections_generator.py:66]: i have not - this is just a slightly rewritten guardrail from what was originally in the python notebook

File: tests/unit/cellguide_pipeline/test_source_collections_generator.py
Comment (joyceyan) [reply tests/unit/cellguide_pipeline/test_source_collections_generator.py:16]: lol ty

**Pull request merged**
